checks/adminbar.php

```diff
@@ -1,41 +1,48 @@
 <?php
 /**
- * This checks, if the admin bar gets hidden by the theme
+ * This checks if the admin bar gets hidden by the theme.
  **/
 class NoHiddenAdminBar implements themecheck {
 	protected $error = array();
 
-		function check( $php_files, $css_files, $other_files ) {
-			$ret = true;
-			checkcount();
-			$php_regex = "/(add_filter(\s*)\((\s*)(\"|')show_admin_bar(\"|')(\s*)(.*))|(([^\S])show_admin_bar(\s*)\((.*))/";
-			$css_regex = "/(#wpadminbar)/";
+	function check( $php_files, $css_files, $other_files ) {
+		$ret = true;
+		checkcount();
+		$php_regex = "/(add_filter(\s*)\((\s*)(\"|')show_admin_bar(\"|')(\s*)(.*))|(([^\S])show_admin_bar(\s*)\((.*))/";
+		$css_regex = "/(#wpadminbar)/";
 
-			//Check php files for filter show_admin_bar and show_admin_bar()
-			foreach ( $php_files as $file_path => $file_content ) {
+		// Check php files for filter show_admin_bar, show_admin_bar_front, and show_admin_bar().
+		foreach ( $php_files as $file_path => $file_content ) {
 
-				$filename = tc_filename( $file_path );
+			$filename = tc_filename( $file_path );
 
-				if ( preg_match( $php_regex, $file_content, $matches ) ) {
-					$this->error[] = sprintf( '<span class="tc-lead tc-required">' . __( 'REQUIRED', 'theme-check').'</span>: ' . __( 'You are not allowed to hide the admin bar.', 'theme-check' ), 
-						'<strong>' . $filename . '</strong>');	
-					$ret = false;			
-				}
+			if ( preg_match( $php_regex, $file_content, $matches ) ) {
+
+				$error = '/show_admin_bar/';
+				$grep  = tc_preg( $error, $file_path );
+
+				$this->error[] = sprintf( '<span class="tc-lead tc-warning">' . __( 'WARNING', 'theme-check' ) . '</span>: ' . __( '%1$s Themes are not allowed to hide the admin bar. This warning must be manually checked.', 'theme-check' ),
+				'<strong>' . $filename . '</strong>' ) . $grep;
 			}
+		}
 
-			//Check CSS Files for #wpadminbar
-			foreach ( $css_files as $file_path => $file_content ) {
+		// Check CSS Files for #wpadminbar.
+		foreach ( $css_files as $file_path => $file_content ) {
 
-				$filename = tc_filename( $file_path );
+			$filename = tc_filename( $file_path );
+			$error    = '/#wpadminbar/';
+			$grep     = tc_preg( $error, $file_path );
 
-				if ( preg_match( $css_regex, $file_content, $matches ) ) {
-					$this->error[] = sprintf( '<span class="tc-lead tc-warning">' . __( 'WARNING', 'theme-check').'</span>: ' . __( 'The theme is using `#wpadminbar`. Hiding the admin bar is not allowed.', 'theme-check' ), 
-						'<strong>' . $filename . '</strong>');		
-				}
+			if ( preg_match( $css_regex, $file_content, $matches ) ) {
+				$this->error[] = sprintf( '<span class="tc-lead tc-warning">' . __( 'WARNING', 'theme-check' ) . '</span>: ' . __( 'The theme is using `#wpadminbar` in %1$s. Hiding the admin bar is not allowed. This warning must be manually checked.', 'theme-check' ),
+				'<strong>' . $filename . '</strong>' ) . $grep;
```

Comment:
$grep will output the entire line, so, it should't be output if the CSS file is minified.

Reply:
You can check for minified files like I did in the theme sniffer: https://github.com/WPTRT/theme-sniffer/blob/dcfd2f181fae0ea17c99e461ebf2bc7be571e64f/src/callback/class-run-sniffer-callback.php#L424

One additional check that comes to mind is to 'parse' a file through `file_get_contents()` function, and get the line length. Since minified files are usually 1 line, but super long, that could be an extra check (if line length longer than 200-300 characters skip).